elements/lisk-p2p/src/peer_pool.ts

```diff
@@ -408,18 +408,31 @@ export class PeerPool extends EventEmitter {
 		peer.send(message);
 	}
 
-	public discoverSeedPeers(): void {
-		const openOutboundSlots = this._getAvailableOutboundConnectionSlots();
+	public discoverFromSeedPeers(): void {
+		const freeOutboundSlots = this.getFreeOutboundSlots();
 
-		if (openOutboundSlots === 0 || this._peerLists.seedPeers.length === 0) {
+		// LIP-0004 re-discovery SeedPeers when Outboundconnection < maxOutboundconnections
+		if (freeOutboundSlots === 0 || this._peerLists.seedPeers.length === 0) {
 			return;
 		}
 
-		const seedPeersForConnection = shuffle(
-			this._peerLists.seedPeers.slice(0, openOutboundSlots),
+		// Looking after existing seed peer connection(s)
+		this._peerLists.seedPeers.forEach(peer => {
+			const isConnectedSeedPeer = this.getPeer(peer.peerId);
+			if (isConnectedSeedPeer) {
+				// tslint:disable-next-line: no-floating-promises
+				(async () => {
+					await isConnectedSeedPeer.discoverPeers();
```

Comment:
if this throws, it would be unhandled rejection, it should be try-catched and ignored or emit error event